nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+                        for (int i = 0; i < numRecords; i++) {
+                            for (RecordField writeRecordField : writeFieldNames) {
+                                final String writeFieldName = writeRecordField.getFieldName();
+                                final Object writeFieldValue;
+                                if (usingSchema) {
+                                    writeFieldValue = generateValueFromRecordField(writeRecordField, faker, nullPercentage);
+                                } else {
+                                    final boolean nullValue;
+                                    if (!context.getProperty(GenerateRecord.NULLABLE_FIELDS).asBoolean() || nullPercentage == 0) {
+                                        nullValue = false;
+                                    } else {
+                                        nullValue = (faker.number().numberBetween(0, 100) <= nullPercentage);
+                                    }
+                                    if (nullValue) {
+                                        writeFieldValue = null;
+                                    } else {
+                                        final String propertyValue = context.getProperty(writeFieldName).getValue();
+                                        writeFieldValue = getFakeData(propertyValue, faker);
+                                    }
+                                }
+
+                                recordEntries.put(writeFieldName, writeFieldValue);
+                            }
+                            record = new MapRecord(recordSchema, recordEntries);
+                            writer.write(record);
+                        }
+
+                        final WriteResult writeResult = writer.finishRecordSet();
+                        attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+                        attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+                        attributes.putAll(writeResult.getAttributes());
+                        recordCount.set(writeResult.getRecordCount());
+                    }
+                } catch (final SchemaNotFoundException e) {
+                    throw new ProcessException(e.getLocalizedMessage(), e);
+                }
+            });
+        } catch (final Exception e) {
+            getLogger().error("Failed to process {}; will route to failure", flowFile, e);
+            if (e instanceof ProcessException) {
+                throw e;
+            } else {
+                throw new ProcessException(e);
+            }
+        }
+
+        flowFile = session.putAllAttributes(flowFile, attributes);
+        session.transfer(flowFile, REL_SUCCESS);
+
+        final int count = recordCount.get();
+        session.adjustCounter("Records Processed", count, false);
+
+        getLogger().info("Successfully generated {} records for {}", count, flowFile);
```

Comment:
Recommend slight wording adjustment, which is more terse, but also reads well enough for `1` record.
```suggestion
        getLogger().info("Generated records [{}] for {}", count, flowFile);
```